src/client/internal.ts

```diff
@@ -246,35 +246,43 @@ export interface ChannelReceiveContextPayload {
 /**
  * Invokes an array of listeners with a given context, allowing us to apply consistent error handling. Will throw an error if > 0 listeners
  * are given, and all fail
+ * The first only *defined* return value is returned
```

Comment:
Can this fit better with the existing comment?

    are given, and all fail. Otherwise, the first *defined* value returned by a listener is returned, or undefined if no defined values are returned.